### PR TITLE
Labb  10

### PR DESCRIPTION
**Pull request by DenisMilkevich**

### CODE REVIEW

File: lecture_notes/10 refactoring/1-not-optimal.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            json_string = points[i].replace("'", "\"")
+            points[i] = json.loads(json_string)
+            points[i]['x'] += offset['x']
+            points[i]['y'] += offset['y']
+
+    return points
+
+
+polyline = [
+    {'x': 0, 'y': 0},
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/1-not-optimal.py:21]:
```suggestion
    { 'x': 0, 'y': 0 },
```

File: lecture_notes/10 refactoring/1-not-optimal.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    {'x': 10, 'y': 10},
+    '{ "x": 20, "y": 20 }',
+    {'x': 30, 'y': 30},
+]
+
+shift({'x': 10, 'y': -5}, polyline)
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/1-not-optimal.py:27]: Improve indentation and formatting

File: lecture_notes/10 refactoring/3 - step.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import json
+
+
+def shift(coords):
+    def move(point):
+        point['x'] += coords['x']
+        point['y'] += coords['y']
+        return point
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/3 - step.py:8]: Try not to change incoming parameters

File: lecture_notes/10 refactoring/3 - step.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const move =
+  ({ dx, dy }) =>
+    ({ x, y }) => ({
+    //Returning result
+      x: x + dx,
+      y: y + dy,
+    });
+
+const coordParse = (item) =>
+  //Cheking for 'True' or 'False'
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/3 - step.js:12]: Thus comments is not needed here

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
Waiting for more optimizations

File: lecture_notes/10 refactoring/1-not-optimal.py (hunk 2 above)
Comment (DenisMilkevich) [reply lecture_notes/10 refactoring/1-not-optimal.py]: Done

**Pull request closed without merging**